hfaerber/Refactor-tractor-Fit-lit
# Dry

@Dunn-Austen @NaomiKC92 

#### What's this PR do?

This PR resolves issue (already closed) #13.  

It also would close another issue, #5 because it refactors the methods dynamically but we need to discuss whether we want the methods that return data for one week to be able to accept a start date for that week.   Currently these are build to accept a number (like 1) for the most recent 7 days or 2 for the 7 days before that. 

This PR will close issue #6 by correcly renaming the method to find the all time stair record. 

This PR removes the unused returnWeek() method in the Activity class that is noted in issue #3 and writes tests for the other two methods noted in issue #3.   This will fix #3. 

#### Where should the reviewer start?

Reviewers should start with the activity file as most changes started there.  All have been updated accordingly in the script file. 

#### How should this be manually tested?
Reviewers can use `npm test` to confirm that all tests are still passing and can use `npm run start` to confirm that all DOM manipulation is still in tact.

#### Any background context you want to provide?

Our next step will likely be determining where methods used by multiple classes should live.  
I'd love your input on this @Dunn-Austen and @NaomiKC92.   
The findUser() method in each class file is actually finding all the related activity/sleep/hydration stats for the specified user, not the user object itself.  I feel like this method and other shared methods might be best suited for a Utility class/file.   Khalid had mentioned one in my fit lit eval and I think it would make more sense than putting everything in User - but that could work too.  Thoughts?

#### What are the relevant tickets?

#3, #5 and #13 are relevant tickets.

#### Screenshots (if appropriate) . N/A

#### Questions:

- Is there a blog post?  N/A
- Does the knowledge base need an update? N/A
- Does this add new dependencies which need to be added?  No
